src/item_factory.cpp

```diff
@@ -710,24 +710,15 @@ void Item_factory::load( islot_gun &slot, JsonObject &jo )
         }
     }
 
-    //Add any built-in mods.
-    if( jo.has_array( "built_in_mods" ) ) {
-    JsonArray jarr = jo.get_array( "built_in_mods" );
-        while( jarr.has_more() ) {
-            std::string temp = jarr.next_string();
-            slot.built_in_mods.push_back( temp );
-        }
+    JsonArray builtmod = jo.get_array( "built_in_mods" );
+    while( builtmod.has_more() ) {
+        slot.built_in_mods.emplace( builtmod.next_string() );
```

Comment:
I've generally favored it as the default because you can never be certain that someone else won't then call your function in a tight loop (granted that would be contrived in the above code). If it can sometimes prevent a move though that means I'll have to reconsider that viewpoint.